test/types/queries.test.ts

```diff
@@ -307,20 +313,34 @@ function autoTypedQuery() {
 }
 
 function gh11964() {
+  class Repository<T extends { id: string }> {
```

Comment:
maybe we can make / have a helper that is a `BasicPojoDocument` for easy / quick / shared access